### ISSUE TITLE
Proper weight decay for Adam optimizer

### ISSUE THREAD

**Issue by novog** [tflearn/tflearn#772]
With a DNN that I am using, I have found that with the Momentum optimizer, a fairly typical weight decay of 0.001 works well. However, when using an Adam optimizer with the same network, I get much better results with a wd in the 10.0 - 100.0 range. I have never heard of anyone using values that large, and they do not work well with the Momentum optimizer. Is it possible that weight decay is not being handled correctly in the Adam optimizer?

I am using tflearn 0.3.1 with tensorflow-gpu 1.1.0.
